### PR TITLE
[processing] port enum and matrix parameter definition widgets to C++

### PR DESCRIPTION
**Pull request by alexbruy**
## Description
Port enum and matrix parameter definition widgets to C++. This allows to use native API to create parameter definition dialogs in modeler.

### CODE REVIEW

**Comment by alexbruy** [comment]
Travis build passed but results were not transmitted to GitHub for some reason.

File: python/gui/auto_generated/processing/qgsprocessingenummodelerwidget.sip.in [hunk 1]
```diff
@@ -0,0 +1,85 @@
+/************************************************************************
+ * This file has been generated automatically from                      *
+ *                                                                      *
+ * src/gui/processing/qgsprocessingenummodelerwidget.h                  *
+ *                                                                      *
+ * Do not edit manually ! Edit header and run scripts/sipify.pl again   *
```
Comment (nyalldawson) [inline python/gui/auto_generated/processing/qgsprocessingenummodelerwidget.sip.in]: We could "#define SIP_NO_FILE" these two -- we don't have to expose them

File: src/gui/processing/qgsprocessingmatrixmodelerwidget.cpp [hunk 2]
```diff
@@ -0,0 +1,162 @@
...
+  for ( int i = 0; i < mModel->rowCount(); ++i )
+    items << new QStandardItem( '0' );
+
+  mModel->appendColumn( items );
+}
+
+void QgsProcessingMatrixModelerWidget::removeColumns()
+{
+  QModelIndexList selected = mTableView->selectionModel()->selectedColumns();
+  std::sort( selected.begin(), selected.end() );
```
Comment (nyalldawson) [inline src/gui/processing/qgsprocessingmatrixmodelerwidget.cpp]: This needs sorting in reverse

File: src/gui/processing/qgsprocessingmatrixmodelerwidget.cpp [hunk 3]
```diff
@@ -0,0 +1,162 @@
...
+  for ( int i = 0; i < mModel->columnCount(); ++i )
+    items << new QStandardItem( '0' );
+
+  mModel->appendRow( items );
+}
+
+void QgsProcessingMatrixModelerWidget::removeRows()
+{
+  QModelIndexList selected = mTableView->selectionModel()->selectedRows();
+  std::sort( selected.begin(), selected.end() );
```
Comment (nyalldawson) [inline src/gui/processing/qgsprocessingmatrixmodelerwidget.cpp]: Also needs reversing

**Review by nyalldawson — APPROVED** [review]

File: python/gui/auto_generated/processing/qgsprocessingenummodelerwidget.sip.in (hunk 1 above)
Comment (alexbruy) [reply python/gui/auto_generated/processing/qgsprocessingenummodelerwidget.sip.in]: Done

File: src/gui/processing/qgsprocessingmatrixmodelerwidget.cpp (hunk 3 above)
Comment (alexbruy) [reply src/gui/processing/qgsprocessingmatrixmodelerwidget.cpp]: Should be fixed now.

File: src/gui/processing/qgsprocessingmatrixmodelerwidget.cpp (hunk 2 above)
Comment (alexbruy) [reply src/gui/processing/qgsprocessingmatrixmodelerwidget.cpp]: Should be fixed now.

**Review by nyalldawson — APPROVED** [review]

**Pull request merged**